code/modules/mob/living/simple_animal/bot/turtle/turtle_cartridge.dm

```diff
@@ -0,0 +1,280 @@
+/datum/stack_item
+	var/line
+	var/skipping
+	var/condition
+
+/datum/stack_item/cmd_if
+/datum/stack_item/cmd_while
+
+
+/obj/item/disk/turtle_cartridge
+	name = "TSL v1.2 Cartridge"
+	desc = "A cartridge for holding Turtle Scripting Language programs. Can be popped into a TurtleBot."
+
+	var/regex/varname_regex = new("^\[a-z\]+$")
```

Comment:
I'm not a regex wizard, I suck with regex and don't know how well (if at all) this will work. But I still have some suggestions:

`var/static/regex/varname_regex = regex("^\[a-z\]+$")`

Or, if you want it to be case insensitive, do:

`var/static/regex/varname_regex = regex("^\[a-z\]+$", "i")`

Reply:
Varnames should only be lowercase

Reply:
Still, you should make it static.